zerver/models.py

```diff
@@ -1055,10 +1055,24 @@ def __str__(self) -> str:
     def is_realm_admin(self) -> bool:
         return self.role == UserProfile.ROLE_REALM_ADMINISTRATOR
 
+    @is_realm_admin.setter
+    def is_realm_admin(self, value: bool) -> None:
+        if value:
+            self.role = UserProfile.ROLE_REALM_ADMINISTRATOR
+        else:
+            self.role = UserProfile.ROLE_MEMBER
+
     @property
     def is_guest(self) -> bool:
         return self.role == UserProfile.ROLE_GUEST
 
+    @is_guest.setter
+    def is_guest(self, value: bool) -> None:
+        if value:
+            self.role = UserProfile.ROLE_GUEST
+        else:
+            self.role = UserProfile.ROLE_MEMBER
```

Comment:
This should avoid changing ROLE_REALM_ADMINISTRATOR -> ROLE_MEMBER by accident.  There's a similar issue with the other one.

I'll just adjust and add tests, since it's a quick change, but FYI.